### DISCUSSION TITLE
Can't build mitmproxy from source due to error

### DISCUSSION THREAD

**Discussion by Botytec** [mitmproxy/mitmproxy#6964] [Q&A]
Every time I follow the instructions [Contributing.md](https://github.com/mitmproxy/mitmproxy/blob/main/CONTRIBUTING.md) for windows, I get this error
Directory '.' is not installable. File 'setup.py' not found.
You are using pip version 10.0.1, however version 24.0 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

i tried going to venv\Scripts in my file explorer and there is no setup.py, am i doing something wrong?

**Comment by mhils** [comment]
You are likely on an outdated Python and pip version. If you follow the instructions closely, you will see that the pip install command needs to be executed in the repository root.
